app/models/events/user_agent_info.rb

```diff
@@ -0,0 +1,36 @@
+class Events::UserAgentInfo < OIDC::BaseModel
+  attribute :installer, :string
+  attribute :device, :string
+  attribute :os, :string
+  attribute :user_agent, :string
+  attribute :implementation, :string
+  attribute :system, :string
+
+  def to_s # rubocop:disable Metrics
+    if installer == "Browser"
+      if device != "Other" && os != "Other" && user_agent != "Other"
+        "#{user_agent} (#{os} on #{device})"
+      elsif device != "Other" && user_agent != "Other"
+        "#{user_agent} (#{device})"
+      elsif os != "Other" && user_agent != "Other"
+        "#{user_agent} (#{os})"
+      elsif user_agent != "Other"
+        user_agent
+      else
+        I18n.t("#{model_name.i18n_key}.unknown_browser")
+      end
```

Comment:
Is this better?

```
installer = user_agent
installer = i18n... if installer == "Other"
os_device = [os, device].reject {|s| s == "Other" }
installer.concat " (#{os_device.join(" on ")})" unless os_device.empty?
```

Similar treatment for the non browser below.

Reply:
I think you'll like this better